plugins/inputs/t128_tank/t128_tank.go

```diff
@@ -121,6 +131,24 @@ func (plugin *T128Tank) checkConfig() error {
 		return fmt.Errorf("topic is a required configuration field")
 	}
 
+	if plugin.IndexFile == "" {
+		if strings.ToLower(plugin.From) == "end" || strings.ToLower(plugin.From) == "eof" {
```

Comment:
This whole addition looks like it can be refactored to be the same meaning in less code.
```
if end:
    index = end
else if start:
    index = start
else:
    if indexfile == "":
         index = end
    else:
 	index = start
```

Reply:
I think that is what i was originally doing but something was not matching up so i did the above. I will change it.

Reply:
I think the confusion is if an index file is specified and has an index, that index should be used. But what is called `plugin.indexValue` is actually the default value and will not be used in the case an index file doesn't have proper data.
1. Rename `indexValue` to `defaultStartingIndex`
2. simplify the logic as Ben suggested